PL2/PL2-core/src/main/java/nl/tudelft/pl2016gr2/core/algorithms/SplitGraphs.java

```diff
@@ -0,0 +1,207 @@
+package nl.tudelft.pl2016gr2.core.algorithms;
+
+import nl.tudelft.pl2016gr2.model.Node;
+import nl.tudelft.pl2016gr2.model.OriginalGraph;
+import nl.tudelft.pl2016gr2.thirdparty.testing.utility.TestId;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * Provides functionality for extracting a subGraph of a sequence graph based on a subset of
+ * genomes.
+ * <p>
+ * The resulting subGraph will provide the following characteristics.
+ * <ul>
+ * <li>The nodes in the subGraph are a subset of the nodes in the original graph.</li>
+ * <li>The nodes in the subGraph will have an identical ID as their counterparts in the original
+ * graph.</li>
+ * <li>The nodes will be newly instantiated, meaning that the subGraph data structure is not backed
+ * by the original graph.</li>
+ * </ul>
+ * </p>
+ *
+ * @author Wouter Smit
+ */
+public class SplitGraphs {
+
+  @TestId(id = "graph_field")
+  private final OriginalGraph mainGraph;
+
+  /**
+   * Instantiates an algorithmic class on a <code>OriginalGraph</code>.
+   * <p>
+   * When {@link #getSubGraph(Collection)} is called, the class will construct a new instance of
+   * the graph containing only the specified genomes.
+   * </p>
+   *
+   * @param mainGraph The graph to split
+   */
+  public SplitGraphs(OriginalGraph mainGraph) {
+    this.mainGraph = mainGraph;
+  }
+
+
+  /**
+   * Builds a subGraph of the <code>mainGraph</code> containing only the nodes that contain at
+   * least one of the specified <code>genomes</code>.
+   *
+   * @param genomes The genomes that should be included in the resulting graph
+   * @return An <code>OriginalGraph</code> that is a subGraph of <code>mainGraph</code>
+   * @throws NoSuchElementException Iff the genome collection is not a subset of the main graph's
+   *                                genomes
+   */
+  public OriginalGraph getSubGraph(Collection<String> genomes) {
+    if (!mainGraph.getGenoms().containsAll(genomes)) {
```

Comment:
We could make it an assert instead, this makes sure that it errors during development (when it fails), but is no longer checked when we build a release, leaving the check out of the final release and thus its performance.

However this depends on the question: Can we ensure that this will never ever happen in release code?

@cbilstra @FarisElghlan